org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/template/java/PostfixCompletionProposalComputer.java

```diff
@@ -137,6 +138,15 @@ public boolean visit(BooleanLiteral node) {
 					return true;
 				}
 
+				@Override
+				public boolean visit(Javadoc node) {
```

Comment:
What does this accomplish ? This visiter logic tries to take the enclosing node (usually for some statement this ends up being the method declaration node), and visit every node to find the largest that occurs closest to the invocation offset. If I include some javadoc around the nodes, I haven't seen a case where this ensures we include something we excluded. Do you have a specific example ?

Reply:
It's trying to get the more accurate node type. The added test case can be an example. We found a EnumConstantDeclaration, and the completion triggered from its JavaDoc child.

So here if we find it's a JavaDoc, record it, and ignore it in `JavaPostfixContext#canEvaluate()`

Reply:
Right. My mistake. Forgot that this part is what gets fed to the selected node in `canEvaluate`.